pdfarranger/pdfarranger.py

```diff
@@ -378,12 +378,14 @@ def generate_booklet(self, _, __, ___):
                  for ref in ref_list]
 
         blank_page_count = 0 if len(pages) % 4 == 0 else 4 - len(pages) % 4
-        for i in range(blank_page_count):
-            self._insert_page(model, pages[0].size, selection)
-            added_page_index = selection[-1].get_indices()[-1] + 1
-            added_page = model.get_value(model.get_iter(added_page_index), 0)
-            pages.append(added_page)
-            model.remove(model.get_iter(added_page_index))
+        if blank_page_count > 0:
+            file = exporter.create_blank_page(self.tmp_dir, pages[0].size)
+            for i in range(blank_page_count):
+                self._insert_pages(model, file, selection)
+                added_page_index = selection[-1].get_indices()[-1] + 1
+                added_page = model.get_value(model.get_iter(added_page_index), 0)
+                pages.append(added_page)
+                model.remove(model.get_iter(added_page_index))
```

Comment:
I could be wrong but I think a ```self.model_lock()``` is needed before any model.remove, and a ```self.model_unlock()``` after. You could also put the lock before the for loop, and unlock somewhere after the loop. (I think there is a small chance that a page could get wrong thumbnail without theese locks.) Other than that I think it looks good.

Reply:
Done, using the context manager.